src/main/java/com/GeorgesServer/app/RequestParser.java

```diff
@@ -1,11 +1,57 @@
 package com.GeorgesServer.app;
 
-import java.net.ServerSocket;
+import java.io.BufferedReader;
+import java.util.HashMap;
 
 public class RequestParser {
 
-    public ClientRequest parse(ServerSocket request) {
-        /* ToDo */
-        return null;
+    private RequestReader requestReader;
+    private String convertedRequest;
+    private String method;
+    private String url;
+    private String httpVersion;
+
+    public RequestParser(RequestReader requestReader) {
+        this.requestReader = requestReader;
+    }
+
+    public HashMap<String, String> parse(BufferedReader reader) {
+        convertedRequest = requestReader.read(reader);
+        setMethod();
```

Comment:
These methods seem weird. It might be better to *build* (hint hint) your parsed request in a different way.